app/components/confirmation-modal/component.js

```diff
@@ -0,0 +1,46 @@
+import Ember from 'ember';
+import layout from './template';
+
+export default Ember.Component.extend({
+  confirmationModalService: Ember.inject.service('confirmation-modal'),
+  layout: layout,
+
+  model: Ember.computed.reads('modal.model'),
+
+  init() {
+    this._super.apply(this, arguments);
+    this.get('confirmationModalService').on('open', (modal) => {
+      this.set('modal', modal);
+      this.set('isOpen', true);
+    });
+  },
```

Comment:
Unsure. Changing to the hook for now though.

![](https://api.monosnap.com/rpc/file/download?id=m9Zk7ZZ0SzOP5Ginr9DoOSIV8oDvlN)